src/repr/src/proto/relation.rs

```diff
@@ -7,5 +7,72 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-//! Protobuf structs mirroring [`crate::relation`] are defined
-//! in [`crate::scalar`].
+//! Protobuf structs mirroring `crate::relation`
+
+pub use super::private::proto_scalar_type::ProtoRecordField;
+pub use super::private::{ProtoColumnName, ProtoColumnType};
+use crate::proto::TryFromProtoError;
+use crate::proto::TryIntoIfSome;
+use crate::{ColumnName, ColumnType};
+
+impl From<&ColumnName> for ProtoColumnName {
+    fn from(x: &ColumnName) -> Self {
+        ProtoColumnName {
+            value: Some(x.0.clone()),
+        }
+    }
+}
+
+impl TryFrom<ProtoColumnName> for ColumnName {
+    type Error = TryFromProtoError;
+
+    fn try_from(x: ProtoColumnName) -> Result<Self, Self::Error> {
+        Ok(ColumnName(x.value.ok_or_else(|| {
+            TryFromProtoError::MissingField("ProtoColumnName::value".into())
+        })?))
+    }
+}
+
+impl From<&ColumnType> for ProtoColumnType {
+    fn from(x: &ColumnType) -> Self {
+        ProtoColumnType {
+            nullable: x.nullable,
+            scalar_type: Some((&x.scalar_type).into()),
+        }
+    }
+}
+
+impl TryFrom<ProtoColumnType> for ColumnType {
+    type Error = TryFromProtoError;
+
+    fn try_from(x: ProtoColumnType) -> Result<Self, Self::Error> {
+        Ok(ColumnType {
+            nullable: x.nullable,
+            scalar_type: x
+                .scalar_type
+                .try_into_if_some("ProtoColumnType::scalar_type")?,
+        })
+    }
+}
+
+impl From<&(ColumnName, ColumnType)> for ProtoRecordField {
```

Comment:
I think this should be in `proto/scalar.rs`, as it is used in the `{Try}From` traits for `ScalarType` which is in defined in `src/scalar.rs`.